src/vs/platform/contextkey/common/scanner.ts

```diff
@@ -0,0 +1,387 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { CharCode } from 'vs/base/common/charCode';
+import { illegalArgument } from 'vs/base/common/errors';
+
+export enum TokenType {
```

Comment:
I would suggest to convert this to a `const enum` and let TS auto-number it for you. Switching to a const enum will lead to TS inlining the numbers in the generated JS code and in the comparisons being a lot faster at runtime. This is also a low hanging perf improvement.

Reply:
interestingly, letting TS to auto-generate values for enum cases for you as below blows up running time from ~10ms to ~200ms

```ts
export const enum TokenType {
	LParen,
	RParen,
...
} 
```

<img width="609" alt="image" src="https://user-images.githubusercontent.com/16353531/219676950-286942a6-e48f-4dfb-b1db-df7a4fa181c1.png">

I will experiment with other ways to speed this up.